gusto/diagnostics.py

```diff
@@ -195,6 +196,67 @@ def compute(self, state):
         return self.field
 
 
+class SphericalComponent(DiagnosticField):
+
+    def __init__(self, name):
+        super().__init__()
+        self.fname = name
+
+    def setup(self, state):
+        if not self._initialised:
+            # check geometric dimension is 3D
+            if state.mesh.geometric_dimension() != 3:
+                raise ValueError('Spherical components only work in 3D!')
+            space = FunctionSpace(state.mesh, "CG", 1)
+            super().setup(state, space=space)
+
+        V = VectorFunctionSpace(state.mesh, "CG", 1)
+        self.x, self.y, self.z = SpatialCoordinate(state.mesh)
+        self.x_hat = Function(V).interpolate(as_vector([Constant(1.0), 0.0, 0.0]))
```

Comment:
Can `Constant` come outside `as_vector`? The zeros are constant too.

Reply:
Done!